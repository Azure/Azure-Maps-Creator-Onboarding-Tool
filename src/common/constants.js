export const PATHS = {
  INDEX: '/',
  VIEW_CONVERSIONS: '/view-conversions',
  CREATE_UPLOAD: '/create-upload',
  PROCESSING: '/processing',
  CREATE_GEOREFERENCE: '/create-georeference',
  REVIEW_CREATE: '/review-create',
  CONVERSION: '/review-create/conversion',
  IMDF_CONVERSION: '/review-create/imdf-conversion',
  PAST_CONVERSION: '/past-conversion',
  CONVERSIONS: '/conversions',
  LAYERS: '/layers',
  LEVELS: '/levels',
  INVALID_PATH: '/*',
};

export const ROUTE_NAME_BY_PATH = {
  [PATHS.INDEX]: 'home',
  [PATHS.CREATE_UPLOAD]: 'Create',
  [PATHS.VIEW_CONVERSIONS]: 'View',
  [PATHS.PROCESSING]: 'processing',
  [PATHS.CREATE_GEOREFERENCE]: 'prepare.drawing.package',
  [PATHS.LAYERS]: 'prepare.drawing.package',
  [PATHS.LEVELS]: 'prepare.drawing.package',
  [PATHS.INVALID_PATH]: 'redirect',
  [PATHS.REVIEW_CREATE]: 'prepare.drawing.package',
  [PATHS.CONVERSION]: 'create.indoor.map',
  [PATHS.IMDF_CONVERSION]: 'convert.indoor.map',
  [PATHS.CONVERSIONS]: 'All conversions',
};

export const PLACES_ROUTE_NAME_BY_PATH = {
  [PATHS.CREATE_GEOREFERENCE]: 'prepare.building.configuration',
  [PATHS.LAYERS]: 'prepare.building.configuration',
  [PATHS.LEVELS]: 'prepare.building.configuration',
  [PATHS.REVIEW_CREATE]: 'prepare.building.configuration',
};

export const TRUNCATE_FRACTION_DIGITS = 8;

export const HTTP_STATUS_CODE = {
  ACCEPTED: 202,
  OK: 200,
  BAD_REQUEST: 400,
  UNAUTHORIZED: 401,
  FORBIDDEN: 403,
  TOO_MANY_REQUESTS: 429,
};

export const isStagingEnv = process.env.REACT_APP_STAGING_ENV === 'true';

export const PLACES_PREVIEW = {
  DEFAULT_LANGUAGE: 'en',
  VERSION: 'places-1.0',
  DESCRIPTION: 'Converted by AzMapsCreatorOnboardingTool',
  STORAGE_RETENTION: 'day',
  SEARCH_PARAMETER: 'placespreview',
};
