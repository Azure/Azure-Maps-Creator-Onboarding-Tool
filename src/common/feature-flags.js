const flags = {
  onboardingEnabled: false,
};

export default flags;