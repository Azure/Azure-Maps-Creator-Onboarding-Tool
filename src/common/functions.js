/**
 * Given a single path name, returns a list of all path names from root to the given path name.
 * Ex. Given '/create/hello', returns ['/', '/create', '/create/hello']
 * @param {String} pathName the path name of the URL
 * @returns {String[]} an array of path names
 */
export const getSplitPaths = (pathName) => {
  if (typeof pathName !== 'string' || pathName === '') { return []; }
  if (pathName === '/') { return ['/']; }

  const pathNameLowerCase = pathName.toLowerCase();
  const splits = pathNameLowerCase.split('/');
  const splitPaths = [];
  for (let i = splits.length - 1; i > 0; i--) {
    let stringBuilder = '';
    for (let j = 1; j <= i; j++) {
      stringBuilder += '/' + splits[j];
    }
    splitPaths.push(stringBuilder);
  }
  splitPaths.push('/');
  splitPaths.reverse();

  return splitPaths;
};