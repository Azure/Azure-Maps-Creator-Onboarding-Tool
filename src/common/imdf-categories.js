export const imdfCategories = [
  'auditorium',
  'brick',
  'classroom',
  'column',
  'concrete',
  'conferenceroom',
  'drywall',
  'elevator',
  'escalator',
  'fieldofplay',
  'firstaid',
  'fitnessroom',
  'foodservice',
  'footbridge',
  'glass',
  'huddleroom',
  'kitchen',
  'laboratory',
  'library',
  'lobby',
  'lounge',
  'mailroom',
  'mothersroom',
  'movietheater',
  'movingwalkway',
  'nonpublic',
  'office',
  'opentobelow',
  'parking',
  'phoneroom',
  'platform',
  'privatelounge',
  'ramp',
  'recreation',
  'restroom',
  'restroom.family',
  'restroom.female',
  'restroom.female.wheelchair',
  'restroom.male',
  'restroom.male.wheelchair',
  'restroom.transgender',
  'restroom.transgender.wheelchair',
  'restroom.unisex',
  'restroom.unisex.wheelchair',
  'restroom.wheelchair',
  'road',
  'room',
  'serverroom',
  'shower',
  'smokingarea',
  'stairs',
  'steps',
  'storage',
  'structure',
  'terrace',
  'theater',
  'unenclosedarea',
  'unspecified',
  'vegetation',
  'waitingroom',
  'walkway',
  'walkway.island',
  'wood',
];
