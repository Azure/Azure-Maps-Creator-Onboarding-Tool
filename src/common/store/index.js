export { conversionSteps, useConversionStore } from './conversion.store';
export { useDissolvedExterior, useGeometryStore } from './geometry.store';
export { useLayersStore } from './layers.store';
export { useLevelsStore } from './levels.store';
export {
  progressBarStepsByKey,
  useCompletedSteps,
  useProgressBarSteps,
  useProgressBarStore,
} from './progress-bar-steps';
export { resetStores } from './reset';
export { LRO_STATUS, useResponseStore } from './response.store';
export { useReviewManifestJson, useReviewManifestStore } from './review-manifest.store';
export { getDomain, useUserStore } from './user.store';
