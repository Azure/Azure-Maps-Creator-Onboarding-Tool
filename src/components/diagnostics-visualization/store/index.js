import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

import rootReducer from '../reducers';

// Middlewares go here
const middleware = {};

const configureStore = (initialState) => createStore(
  rootReducer,
  initialState,
  // Redux Dev Tools Middleware
  composeWithDevTools(applyMiddleware(...middleware)),
);

// pass an optional param to rehydrate state on app start
const store = configureStore();

// export store singleton instance
export default store;
