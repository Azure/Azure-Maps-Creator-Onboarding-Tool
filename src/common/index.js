import { PATHS, ROUTE_NAME_BY_PATH } from './constants';

export { PATHS, ROUTE_NAME_BY_PATH };