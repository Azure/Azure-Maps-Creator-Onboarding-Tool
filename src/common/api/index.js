// The following API is designed to be used only by the onboarding tool and is not supported for any other use.
import { getEnvs } from 'common/functions';
import { useUserStore } from '../store/user.store';

export const uploadFile = (file) => {
  const { geography, subscriptionKey } = useUserStore.getState();
  const url = `${getEnvs()[geography].URL}/manifest?api-version=2.0&subscription-key=${subscriptionKey}`;

  return fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/zip',
    },
    body: file,
  });
};

export const fetchFromLocation = (location) => {
  const { subscriptionKey } = useUserStore.getState();
  return fetch(`${location}&subscription-key=${subscriptionKey}`, {
    method: 'GET',
  });
};

export const deleteFromLocation = (fetchUrl) => {
  const { subscriptionKey } = useUserStore.getState();
  fetch(`${fetchUrl}&subscription-key=${subscriptionKey}`, {
    method: 'DELETE',
  });
};

export const fetchAddress = (address) => {
  const { geography, subscriptionKey } = useUserStore.getState();
  return fetch(`${getEnvs()[geography].URL}/search/address/json?subscription-key=${subscriptionKey}&api-version=1.0&query=${address}&limit=1`)
    .then((res) => res.json());
};