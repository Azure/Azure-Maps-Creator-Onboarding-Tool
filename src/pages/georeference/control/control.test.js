import { fireEvent, render, screen } from '@testing-library/react';
import flushPromises from 'flush-promises';

import { useGeometryStore, useUserStore } from 'common/store';
import GeoreferenceControl from './controlClass';
import Control from './control';

describe('GeoreferenceControl class', () => {
  let control, container;
  const map = { map: 'map' };

  beforeEach(() => {
    useGeometryStore.setState({
      anchorPoint: {
        coordinates: [0,0],
        angle: 111,
      },
    });
    useUserStore.setState({
      geography: 'EU',
      subscriptionKey: 'subKey',
    });

    jest.useFakeTimers();
    control = new GeoreferenceControl();
    container = control.onAdd(map);
    // added cause without it useEffect hook wasn't triggered properly and value was not updated correctly
    jest.advanceTimersByTime(0);
  });

  it('onAdd should return container', () => {
    expect(container).toMatchSnapshot();
    expect(control.map).toBe(map);
  });

  it('onRemove should delete container and map', () => {
    control.onRemove();
    expect(true).toBe(true);
    expect(control.container).toBe(null);
    expect(control.map).toBe(null);
  });
});

describe('Control', () => {
  it('should render Control', () => {
    const view = render(<Control />);
    expect(view).toMatchSnapshot();
  });

  it('should center map to set coordinates', () => {
    const map = {
      setCamera: jest.fn(),
    };
    render(<Control map={map} />);

    const lngInput = screen.getByPlaceholderText('longitude');
    const latInput = screen.getByPlaceholderText('latitude');

    fireEvent.change(lngInput, {
      target: {
        value: '123.123',
      },
    });
    fireEvent.change(latInput, {
      target: {
        value: '321.321',
      },
    });
    fireEvent.keyUp(lngInput, {
      code: 'Enter',
      charCode: 13,
    });

    expect(map.setCamera).not.toBeCalled();

    fireEvent.change(latInput, {
      target: {
        value: '50.99',
      },
    });
    fireEvent.keyPress(lngInput, {
      code: 'Enter',
      charCode: 13,
    });
    expect(map.setCamera).toBeCalledWith({
      center: [123.123, 50.99],
    });
  });

  it('should search address', async () => {
    global.fetch.mockReturnValue(
      Promise.resolve({
        json: () => ({
          results: [{
            position: {
              lon: '12.34',
              lat: '56.78',
            }
          }],
        }),
      })
    );
    const map = { setCamera: jest.fn() };
    render(<Control map={map} />);
    const searchAddressInput = screen.getByPlaceholderText('search.address');
    fireEvent.change(searchAddressInput, {
      target: {
        value: '742 Evergreen Terrace',
      },
    });
    fireEvent.keyPress(searchAddressInput, {
      code: 'Enter',
      charCode: 13,
    });

    await flushPromises();

    expect(global.fetch).toHaveBeenCalledWith('https://eu.atlas.microsoft.com/search/address/json?subscription-key=subKey&api-version=1.0&query=742 Evergreen Terrace&limit=1');
    expect(map.setCamera).toHaveBeenCalledWith({
      center: [12.34, 56.78],
      zoom: 16,
    });
  });

  it('should show error message', async () => {
    global.fetch.mockReturnValue(
      Promise.reject('something really bad happened')
    );
    const map = { setCamera: jest.fn() };
    const view = render(<Control map={map} />);
    const searchAddressInput = screen.getByPlaceholderText('search.address');
    fireEvent.change(searchAddressInput, {
      target: {
        value: '742 Evergreen Terrace',
      },
    });
    fireEvent.keyPress(searchAddressInput, {
      code: 'Enter',
      charCode: 13,
    });

    await flushPromises();

    expect(global.fetch).toHaveBeenCalledWith('https://eu.atlas.microsoft.com/search/address/json?subscription-key=subKey&api-version=1.0&query=742 Evergreen Terrace&limit=1');
    expect(view).toMatchSnapshot();
  });
});