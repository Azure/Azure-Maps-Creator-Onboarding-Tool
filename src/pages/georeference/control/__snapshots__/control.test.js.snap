// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeoreferenceControl class onAdd should return container 1`] = `
.emotion-0 {
  position: relative;
}

.emotion-1 {
  width: 2rem;
  height: 2rem;
  background-color: white;
  color: black;
  border: none;
  border-radius: 2px;
  box-shadow: rgb(0 0 0 / 16%) 0 0 4px;
  cursor: pointer;
  display: none;
}

.emotion-1:hover {
  background-color: white;
  color: black;
}

.emotion-1:active {
  background-color: #f1f1f1;
  color: black;
}

.emotion-2 {
  background-color: white;
  font-size: 0.813rem;
}

.emotion-3 {
  font-weight: 600;
  padding: 0.3125rem 0.5rem 0.3125rem 0.875rem;
  border-bottom: 1px solid #F3F2F1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  border: none;
  background: none;
  cursor: pointer;
  color: black;
}

.emotion-4:hover {
  background-color: white;
  color: black;
}

.emotion-4:active {
  background-color: #f1f1f1;
  color: black;
}

.emotion-5 {
  padding: 0.7rem 0.875rem;
  border-bottom: 1px solid #F3F2F1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 0;
}

.emotion-6 {
  padding: 0.7rem 0.875rem;
  border-bottom: 1px solid #F3F2F1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-7 {
  margin-top: 0.5rem;
  height: 1.5rem;
  font-size: 0.813rem;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: flex-end;
  align-self: flex-end;
}

.emotion-8 {
  padding: 0.7rem 0.875rem;
}

.emotion-9 {
  margin-bottom: 0.2rem;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-11 {
  width: 10rem;
  height: 1.5rem;
}

.emotion-12 {
  width: 3rem;
  margin-left: 0.25rem;
  text-align: center;
}

<div
  class="azure-maps-control-container"
>
  <div
    class="emotion-0"
    data-testid="map-control"
  >
    <button
      aria-label="hide.control"
      class="ms-Button ms-Button--icon emotion-1 root-110"
      data-is-focusable="true"
      type="button"
    >
      <span
        class="ms-Button-flexContainer flexContainer-111"
        data-automationid="splitbuttonprimary"
      >
        <i
          aria-label="hide.control"
          class="root-120"
          data-icon-name="Search"
          role="img"
        >
          
        </i>
      </span>
    </button>
    <div
      class="emotion-2"
    >
      <div
        class="emotion-3"
      >
        <span>
          position.building.footprint
        </span>
        <button
          aria-label="toggle.control"
          class="ms-Button ms-Button--icon emotion-4 root-110"
          data-is-focusable="true"
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-111"
            data-automationid="splitbuttonprimary"
          >
            <i
              aria-hidden="true"
              class="root-123"
              data-icon-name="BackToWindow"
            >
              
            </i>
          </span>
        </button>
      </div>
      <div
        class="emotion-5"
      >
        <div
          class="ms-ChoiceFieldGroup root-124"
        >
          <div
            role="radiogroup"
          >
            <div
              class="ms-ChoiceFieldGroup-flexContainer"
            >
              <div
                class="ms-ChoiceField root-125"
              >
                <div
                  class="ms-ChoiceField-wrapper"
                >
                  <input
                    checked=""
                    class="ms-ChoiceField-input input-126"
                    id="ChoiceGroup6-address"
                    name="ChoiceGroup6"
                    type="radio"
                  />
                  <label
                    class="ms-ChoiceField-field is-checked field-127"
                    for="ChoiceGroup6-address"
                  >
                    <span
                      class="ms-ChoiceFieldLabel"
                      id="ChoiceGroupLabel7-address"
                    >
                      search.by.building.address
                    </span>
                  </label>
                </div>
              </div>
              <div
                class="ms-ChoiceField root-125"
              >
                <div
                  class="ms-ChoiceField-wrapper"
                >
                  <input
                    class="ms-ChoiceField-input input-126"
                    id="ChoiceGroup6-coordinates"
                    name="ChoiceGroup6"
                    type="radio"
                  />
                  <label
                    class="ms-ChoiceField-field field-132"
                    for="ChoiceGroup6-coordinates"
                  >
                    <span
                      class="ms-ChoiceFieldLabel"
                      id="ChoiceGroupLabel7-coordinates"
                    >
                      search.by.lon.lat
                    </span>
                  </label>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="emotion-6"
      >
        <div
          class="ms-TextField root-136"
        >
          <div
            class="ms-TextField-wrapper"
          >
            <div
              class="ms-TextField-fieldGroup fieldGroup-137"
            >
              <input
                aria-invalid="false"
                class="ms-TextField-field field-138"
                id="TextField8"
                placeholder="search.address"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <button
          aria-disabled="true"
          class="ms-Button ms-Button--primary is-disabled emotion-7 root-147"
          data-is-focusable="false"
          disabled=""
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-111"
            data-automationid="splitbuttonprimary"
          >
            <span
              class="ms-Button-textContainer textContainer-112"
            >
              <span
                class="ms-Button-label label-149"
                id="id__13"
              >
                search
              </span>
            </span>
          </span>
        </button>
      </div>
      <div
        class="emotion-8"
      >
        <div
          class="emotion-9"
        >
          rotation.in.degrees
        </div>
        <div
          class="emotion-10"
        >
          <div
            class="emotion-11"
          >
            <div
              class="ms-Slider ms-Slider-enabled ms-Slider-row root-151"
            >
              <label
                class="ms-Label titleLabel-162"
                for="Slider16"
              />
              <div
                class="ms-Slider-container container-153"
              >
                <div
                  aria-disabled="false"
                  aria-valuemax="360"
                  aria-valuemin="-360"
                  aria-valuenow="111"
                  aria-valuetext="111"
                  class="ms-Slider-slideBox slideBox-154"
                  data-is-focusable="true"
                  id="Slider16"
                  role="slider"
                  tabindex="0"
                >
                  <div
                    class="ms-Slider-line line-156"
                  >
                    <span
                      class="ms-Slider-thumb thumb-155"
                      style="left: 65.41666666666667%;"
                    />
                    <span
                      class="lineContainer-157 ms-Slider-inactive inactiveSection-159"
                      style="width: 0%;"
                    />
                    <span
                      class="lineContainer-157 ms-Slider-active activeSection-158"
                      style="width: 65.41666666666667%;"
                    />
                    <span
                      class="lineContainer-157 ms-Slider-inactive inactiveSection-159"
                      style="width: 34.58333333333333%;"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ms-TextField emotion-12 root-136"
          >
            <div
              class="ms-TextField-wrapper"
            >
              <div
                class="ms-TextField-fieldGroup fieldGroup-163"
              >
                <input
                  aria-invalid="false"
                  aria-label="rotation.in.degrees"
                  autocomplete="off"
                  class="ms-TextField-field field-164"
                  id="TextField17"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
